st3/sublime_lib/pure_resource_path.py

```diff
@@ -0,0 +1,103 @@
+import posixpath
+from functools import total_ordering
+
+
+__all__ = ['PureResourcePath']
+
+
+@total_ordering
+class PureResourcePath():
+    """A representation of a Sublime Text resource path. Inspired by pathlib."""
+
+    def __init__(self, *pathsegments):
+        self._parts = tuple(
+            part
+            for segment in pathsegments if segment
+            for part in posixpath.normpath(str(segment)).split('/')
+        )
+
+    def __hash__(self):
+        return hash(self.parts)
+
+    def __repr__(self):
+        return "%s(%s)" % (type(self).__name__, repr(str(self)))
```

Comment:
Can also use `self.__class__` instead of `type(self)`. No idea about the benefits but I always use the former.

```suggestion
        return "{}({!r})".format(self.__class__.__name__, str(self))
```

Reply:
Done.